pkg/cookies/csrf.go

```diff
@@ -73,15 +73,23 @@ func NewCSRF(opts *options.Cookie, codeVerifier string) (CSRF, error) {
 
 // LoadCSRFCookie loads a CSRF object from a request's CSRF cookie
 func LoadCSRFCookie(req *http.Request, opts *options.Cookie) (CSRF, error) {
-
 	cookieName := GenerateCookieName(req, opts)
 
-	cookie, err := req.Cookie(cookieName)
-	if err != nil {
-		return nil, err
+	cookies := req.Cookies()
+	for _, cookie := range cookies {
+		if cookie.Name != cookieName {
+			continue
+		}
+
+		csrf, err := decodeCSRFCookie(cookie, opts)
+		if err != nil {
+			continue
+		}
```

Comment:
we still don't have logging levels otherwise i would consider adding a debug log here 🤔

Reply:
Yes, I would also like to have a log at this point.
I think we can also add a log here without a log level as this error should not occur so often.